networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/rpc/RpcHandler.java

```diff
@@ -59,14 +62,20 @@ public class RpcHandler<
   private static final Duration TIMEOUT = Duration.ofSeconds(5);
   private static final Logger LOG = LogManager.getLogger();
 
-  private final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod;
   private final AsyncRunner asyncRunner;
+  private final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod;
+  private final Semaphore concurrentRequestsSemaphore;
+  private final Duration concurrentRequestsQueueTimeout;
 
   public RpcHandler(
       final AsyncRunner asyncRunner,
-      final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod) {
+      final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod,
+      final int maxConcurrentRequests,
+      final Duration concurrentRequestsQueueTimeout) {
     this.asyncRunner = asyncRunner;
     this.rpcMethod = rpcMethod;
+    concurrentRequestsSemaphore = new Semaphore(maxConcurrentRequests);
```

Comment:
🤔 ... this is a blocking approach, with a tight timeout... should we have an actual async queue here?

Reply:
Yes, I realised that, actually I am working on a different idea right now. Should be done soon.

Reply:
Changed the approach to use `ThrottlingTaskQueue` which makes much more sense